dashboard/app/controllers/programming_classes_controller.rb

```diff
@@ -58,6 +80,16 @@ def show
     @programming_environment_categories = @programming_class.programming_environment.categories_for_navigation
   end
 
+  def destroy
```

Comment:
Thank you for adding this!

Reply:
Can you add an `after_destroy` hook to ProgrammingClass, like the one for [ProgrammingExpression](https://github.com/code-dot-org/code-dot-org/blob/1a0537466368af22c630ebef5a0363cb2b4a57c0/dashboard/app/models/programming_expression.rb#L35) (you'll need to write the `remove_serialization` function as well but that you can probably just copy paste from https://github.com/code-dot-org/code-dot-org/blob/1a0537466368af22c630ebef5a0363cb2b4a57c0/dashboard/app/models/programming_expression.rb#L286)

Reply:
nice catch- thank you! This is done and remove_serialization was actually already in there. You're so right about most of these files being duplicated...

Reply:
lolol I 100% might've copy pasted that function. There are conversations about what to do with models like this that share some but not all code but we seem to have several different patterns and not settled on one :( I'll keep this in mind if/when I do my laundry list of clean up tasks 😅

Reply:
Totally- that's how I did a lot of this PR! Not to mention- I just added a bunch of tests and essentially copied them over from the expressions_controller.